enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/ServiceDiscoveryEnricher.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.enricher.fabric8;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathFactory;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+import io.fabric8.maven.core.util.Configs;
+import io.fabric8.maven.core.util.MapUtil;
+import io.fabric8.maven.enricher.api.BaseEnricher;
+import io.fabric8.maven.enricher.api.EnricherContext;
+import io.fabric8.maven.enricher.api.Kind;
+
+public class ServiceDiscoveryEnricher extends BaseEnricher {
```

Comment:
Is it about a general service discovery enricher, or only for camel-rest-dsl applications ? AFAIS you can use it only if a bean context file with /beans/camelContext/restConfiguration is found (only the 'discoverable' = true is set).

Wouldn't it make also sense to allow discoverable to be specified ? Or, when description-path and path is explicitely configured don't even check for camel rest dsl but just take this over.

But even then, what about calling it '3ScaleServiceDiscoveryEnricher' ?

Reply:
I think it makes sense to view it as a general service discovery enricher, it for now it only supports Camel DSL.

Yeah I think one should be able to supply the 'discoverable', will add that.

Classes can't start with a '3', so I went for the more generic.